### ISSUE TITLE
Images are displayed on Android but not macOS

### ISSUE THREAD

**Issue by xxyxxyxyx1** [laurent22/joplin#5954]
## Description

I have notes that contain images, encrypted and synced using WebDAV, with Android 11 and macOS 11.6.2 clients. 
Everything is working fine on Android. 
On macOS I do not see any images in my notes. 
I have tried disabling all plugins and restarted Joplin and my computer. The problem has persisted for some days now. AFAIK, my configuration has not been modified (except perhaps by automatic software updates) since things were working. 

Example markdown from a note created months ago, which displayed properly for some time after creation:
```
LMCT:
![7ea36d11d7a0665de6f001d2fb93f558.png](:/a872090ba4874e1d9e3be934a554417d)
```
Screenshot of the Joplin app:
<img width="361" alt="image" src="https://user-images.githubusercontent.com/25559220/147982602-1d11beb6-6312-4da8-8778-d56125a667dd.png">

The folder `~/.config/joplin-desktop/resources` contains many 9-byte files with image extensions. However, I do not see files matching the filenames in my markdown notes. For example, there is no file named `7ea36d11d7a0665de6f001d2fb93f558.png` in this folder. 

On Android, I am able to view this note with the image as expected. 

## Environment

Joplin version: 2.6.10
Platform: macOS
OS specifics: 

Joplin 2.6.10 (prod, darwin)

Client ID: 1e7bbc2addd2462e880704714ad65535
Sync Version: 3
Profile Version: 41
Keychain Supported: Yes

Revision: 98fba37

## Steps to reproduce

1. Select a note containing images on macOS desktop app

## Describe what you expected to happen

Images are displayed

## Logfile
Logfile is attached, with username redacted
[joplin-20220103-1641245376379.log](https://github.com/laurent22/joplin/files/7804104/joplin-20220103-1641245376379.log)

**Comment by xxyxxyxyx1** [comment]
I have also tried `Delete local data and re-download from sync target`, but this did not fix the problem.

**Comment by xxyxxyxyx1** [comment]
I additionally tried exporting everything as JEX and re-importing. This did not fix my problem. I also found errors in the Note Attachments table. 

It seems that at some point, my images all got corrupted for an unknown reason. On Android, things appeared to work only because the old uncorrupted images remained _only on that device_, despite repeated syncing. When I deleted my app data and re-synced from WebDAV, the images were broken as on macOS. 

I also saw various errors similar to these in both macOS and Android:
`"joplin" PUT  Unknown error 2 (502): <html><body><h1>502 Bad Gateway</h1> The server returned an invalid or incomplete response.`
`"joplin" PUT .resource/3ec52bbba4c04b8a8e064a9e4fa20411: Unknown error 2 (502): <html><body><h1>502 Bad Gateway</h1> The server returned an invalid or incomplete response.`

In the end, I exported everything as Markdown and manually removed all resource references from the markdown files using a text editor. I then deleted `~/.config/joplin-desktop` and started over with a new database and new encryption password, then imported the edited markdown export. Luckily, I recovered many images for manual re-insertion using a backup of my WebDAV folder that was made prior to enabling encryption in Joplin.
